pytest_robotframework/__init__.py

```diff
@@ -641,6 +642,94 @@ def pre_rebot_modifier(obj: _T_SuiteVisitor) -> _T_SuiteVisitor:
     return obj
 
 
+class AssertOptions:
+    """pass this as the second argument to an `assert` statement to customize how it appears in the
+    robot log.
+
+    by default, an `assert` statement will display in the robot log as long as the following
+    conditions are met:
+    - it is not inside a `hide_asserts_from_robot_log` context manager
+    - the `enable_assertion_pass_hook` pytest option is enabled
+      (see [enabling pytest assertions in the robot log](https://github.com/DetachHead/pytest-robotframework/#enabling-pytest-assertions-in-the-robot-log)).
+    - pytest is not run with the `--no-asserts-in-robot-log` argument
+
+    :param log:
+        whether to display the assertion as a keyword in the robot log. by default, an `assert`
+        statement will display in the robot log as long as the conditions above are met.
+
+        normally, the asserted expression is displayed as the argument to the keyword, but setting
+        this value to a `str` will display a custom message instead.
+        expression
+
+    :param fail_msg:
+        optional description for the `assert` statement. equivalent to a normal `assert`
+        statement's second argument
+
+    example:
+    -------
+    .. code-block:: python
+    # never displays in the robot log:
+    assert foo == bar, AssertOptions(log=False)
+
+    # always displays in the robot log (as long as the `enable_assertion_pass_hook` pytest option is
+    # enabled):
+    assert foo == bar, AssertOptions(log=True)
+
+    # customizes the argument for the `assert` keyword in the log:
+    assert foo == bar, AssertOptions(log="hello")
+
+    # displays in the robot log as long only if all 3 conditions mentioned above are true:
+    assert foo == bar
+    """
+
+    def __init__(self, *, log: bool | str, fail_msg: str | None = None) -> None:
+        super().__init__()
+        if isinstance(log, str):
+            self.show_in_log = True
+            self.description = log
+        else:
+            self.show_in_log = log
+            self.description = None
+
+        self.fail_msg = fail_msg
+
+    @override
+    def __repr__(self) -> str:
+        """make the custom fail message appear in the call to `AssertionError`"""
+        return self.fail_msg or ""
+
+
+_hide_asserts_context_manager_key = StashKey[bool]()
+
+
+@contextmanager
+def hide_asserts_from_robot_log() -> Iterator[None]:
+    """context manager for hiding multiple `assert` statements from the robot log. note that
+    `assert` statements using `AssertOptions(log=True)` take precedence.
```

Comment:
`AssertOptions(log="asdf")` too which is kinda weird. maybe `log` and `description` should be 2 separate arguments